src/evaluate.jl

```diff
@@ -151,52 +109,57 @@ function evaluate!{T<:Number}(x::Array{Taylor1{TaylorN{T}},1}, δt::T,
     nothing
 end
 
-## Evaluates HomogeneousPolynomials and TaylorN on a val of the nv variable
-## using Horner's rule on the nv variable
-function horner{T<:Number,S<:NumberNotSeriesN}(a::HomogeneousPolynomial{T},
-        b::Tuple{Int,S} )
-    nv, val = b
-    numVars = get_numvars()
-    @assert 1 <= nv <= numVars
+"""
+    evaluate(a, vals)
+
+Evaluate a `HomogeneousPolynomial` polynomial at `vals`.
+"""
+function evaluate{T<:Number,S<:TaylorSeries.NumberNotSeriesN}(a::HomogeneousPolynomial{T},
+        vals::Array{S,1} )
+    @assert length(vals) == get_numvars()
+
+    num_vars = get_numvars()
+    ct = TaylorSeries.coeff_table[a.order+1]
     R = promote_type(T,S)
-    @inbounds indTb = coeff_table[a.order+1]
-    suma = TaylorN(zero(R), a.order)
-
-    # Horner's rule on the nv variable
-    for ord = a.order : -1 : 0
-        suma_ord = TaylorN(zero(R), a.order)
-        posOrd = order_posTb(a.order, nv, ord)
-        neworder = a.order-ord
-        for pos in posOrd
-            c = a[pos]
-            iIndices = copy(indTb[pos])
-            iIndices[nv] = 0
-            kdic = in_base(get_order(), iIndices)
-            newpos = pos_table[neworder+1][kdic]
-            zhp = HomogeneousPolynomial([zero(R)], neworder)
-            zhp[newpos] = a[pos]
-            suma_ord += TaylorN(zhp, a.order)
-        end
-        if ord == a.order
-            suma += suma_ord
-        else
-            suma = suma*val + suma_ord
+    suma = zero(R)
+
+    for (i,a_coeffs) in enumerate(a.coeffs)
+        tmp = vals[1]^(ct[i][1])
```

Comment:
It just access each of the `HomogeneousPolynomial`s of `a` just once. I'm not sure if it's the most efficient way to do it when `vals` is of type `Array{Taylor1{S}}`, but when `vals` is just an array of numbers it's just number operations, which is pretty efficient.